test/unit/math/rev/functor/util_algebra_solver.hpp

```diff
@@ -200,26 +189,7 @@ inline void error_conditions_test(const F& f,
   x_bad_inf << inf, 1, 1;
   EXPECT_THROW_MSG(
       general_algebra_solver(is_newton, f, x_bad_inf, y, dat, dat_int),
-      std::domain_error,
-      "algebra_solver: initial guess[1] is inf, but must "
-      "be finite!");
-
-  typedef Eigen::Matrix<T, Eigen::Dynamic, 1> matrix;
-  matrix y_bad_inf(3);
-  y_bad_inf << inf, 1, 1;
-  EXPECT_THROW_MSG(
-      general_algebra_solver(is_newton, f, x, y_bad_inf, dat, dat_int),
-      std::domain_error,
-      "algebra_solver: parameter vector[1] is inf, but must "
-      "be finite!");
-
-  std::vector<double> dat_bad_inf(1);
-  dat_bad_inf[0] = inf;
-  EXPECT_THROW_MSG(
-      general_algebra_solver(is_newton, f, x, y, dat_bad_inf, dat_int),
-      std::domain_error,
-      "algebra_solver: continuous data[1] is inf, but must "
-      "be finite!");
```

Comment:
I got rid of the check that makes sure the values of the variadic argument are finite, so I got rid of the tests. We've had a conversation about this for the ODEs: https://github.com/stan-dev/math/issues/2406

Can add them back if necessary. @charlesm93 so you know

Reply:
My understanding is that the only change that would result from this is that we need to explicitly add documentation warning the user not to pass infinite values. Is that right?

Reply:
Let's just keep this as it is for now.